src/utilities.cc

```diff
@@ -281,14 +281,32 @@ void SetCommandLine() {
 #endif
 }
 
+/*******************************************************************************
+ * Utility function to format socket information.
+ *******************************************************************************/
+void reportEndpoint(uv_handle_t* h, struct sockaddr* addr, const char* prefix,
+                    std::ostringstream& out) {
+  uv_getnameinfo_t endpoint;
+  if (uv_getnameinfo(h->loop, &endpoint, nullptr, addr, NI_NUMERICSERV) == 0) {
+    out << prefix << endpoint.host << ":" << endpoint.service;
+  } else {
+    char host[INET6_ADDRSTRLEN];
+    int family = addr->sa_family;
+    if (uv_inet_ntop(family, addr, host, sizeof(host)) == 0) {
```

Comment:
Leaving a note here so I don't forget. This needs to be fixed up as `addr` should either be `sockaddr_in->sin_addr` (IPv4) or `sockaddr_in6->sin6_addr` (IPv6) and not `addr`.